### PR TITLE
feat(Footer): Small refactor for better customization

### PR DESCRIPTION
**Pull request by bentleyvk**
Some changes in preparation for https://github.com/iTwin/iTwinUI-react/issues/705.

 - I mainly just flattened the structure and added some classes where needed. 
 - I made footer item to have `display: flex` so that it would be easier to add icons.
 - Moved separator to a separate `li` (same as in Breadcrumbs).

### CODE REVIEW

**Comment by mayank99** [comment]
@bentleyvk Do you have a plan for how we will use these changes in react to allow customizing the icon?

File: packages/itwinui-css/backstop/tests/footer.html [hunk 1]
```diff
@@ -41,54 +41,54 @@ <h1>Footer</h1>
...
+        <li class="iui-legal-footer-item">
           ©&nbsp;
           <script>
             document.write(new Date().getFullYear());
           </script>
           &nbsp;Bentley Systems, Incorporated
-          <span class="iui-separator"></span>
         </li>
-        <li>
+        <li class="iui-legal-footer-separator"></li>
```
Comment (FlyersPh9) [inline packages/itwinui-css/backstop/tests/footer.html:52]: You might be able to remove this element & class and instead use a pseudo element.

**Comment by bentleyvk** [comment]
> @bentleyvk Do you have a plan for how we will use these changes in react to allow customizing the icon?

Yup, I first started to work on React and then I saw that CSS changes are necessary. Footer will accept `children` prop and I plan to expose sub-component `Footer.Item` and maybe `Footer.Separator`.

File: packages/itwinui-css/backstop/tests/footer.html (hunk 1 above)
Comment (bentleyvk) [reply packages/itwinui-css/backstop/tests/footer.html:52]: I was thinking about that but idk, I thought if someone wants to use custom element as footer item, then having separator is kind of useful.

File: packages/itwinui-css/backstop/tests/footer.html [hunk 2]
```diff
@@ -99,9 +99,9 @@ <h1>Footer</h1>
       </ul>
 
       <!-- This section should only appear when signed in with a Bentley Systems account! -->
-      <span class="iui-legal-footer-developer">
+      <div class="iui-legal-footer-developer">
```
[the remark this answers is not in the document]
Comment (bentleyvk) [reply packages/itwinui-css/backstop/tests/footer.html:102]: Currently there is no way in React to achieve this, except by placing some div under footer with negative margin.

**Review by veekeys — APPROVED** [review]

**Review by gretanausedaite — APPROVED** [review]

File: packages/itwinui-css/backstop/tests/footer.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/tests/footer.html:102]: I just checked 6 Bentley web products, only Connect uses that area, no others do.  Might be safe to remove.

**Review by FlyersPh9 — APPROVED** [review]

**Review by mayank99 — APPROVED** [review]
CSS looks good but we should decide between `aria-hidden` and `role=separator` for the `<Footer.Separator>` so that it doesn't get announced as a list item. I think `aria-hidden` makes more sense because these separators are not really meaningful. This is also why I like @FlyersPh9's suggestion to use a pseudo-element with a border.

**Pull request merged**